### PR TITLE
#8: size() method

### LINKED ISSUE #8: Find out what `size()` returns in scalar UDFs
## Situation

In "Set" scripts it is clear what the `size()` method returns. But we are not sure what exactly `size()` produces and if that value has any relevance for the users. If it does not, it should always return 0 instead &mdash; which is a change that then belongs into the UDF client.

## Acceptance Criteria

* Integration test exists that test `size()` in a scalar script
* Meaning of the return value documented

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #8

### CODE REVIEW

File: doc/changes/changes_1.0.2.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Exasol UDF API for Java 1.0.2, released 2023-??-??
```
Comment (kaklakariada) [inline doc/changes/changes_1.0.2.md:1]: No release planned?

File: doc/changes/changes_1.0.2.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Code name: `size()` method
+
+## Summary
+
+In this release we documented the `size()` method and added integration tests for it.
+
+We also updated dependencies.
```
Comment (kaklakariada) [inline doc/changes/changes_1.0.2.md:9]:
```suggestion
We also updated test dependencies.
```

File: src/test/java/com/exasol/test/JavaUdfIT.java [hunk 3]
```diff
@@ -150,13 +151,39 @@ void testGetTimestampFromSetScript() {
...
+                table().row(date + " " + time));
+    }
+
+    private static void assertQueryResult(final String sql, final ResultSetStructureMatcher.Builder rowMatcher) {
         try (final Statement statement = connection.createStatement();
-                final ResultSet result = statement
-                        .executeQuery("SELECT " + fullyQualifiedScriptName + "(T.V)" + "FROM VALUES (TO_TIMESTAMP('"
-                                + date + "T" + time + "Z', 'YYYY-MM-DDTHH24:MI:SS.FF3Z')) AS T(V)")) {
-            assertThat(result, table().row(date + " " + time).matches());
+             final ResultSet result = statement.executeQuery(sql)) {
```
Comment (kaklakariada) [inline src/test/java/com/exasol/test/JavaUdfIT.java:161]:
```suggestion
            final ResultSet result = statement.executeQuery(sql)) {
```
Check formatting

File: src/test/java/com/exasol/test/JavaUdfIT.java [hunk 4]
```diff
@@ -150,13 +151,39 @@ void testGetTimestampFromSetScript() {
...
-}
+
+    @Test
+    void testGetSizeFromScalarScript() {
+        final String scriptName = "SIZE_IN_SCALAR_CONTEXT";
+        final String fullyQualifiedScriptName = getFullyQualifiedScriptName(schema, scriptName);
+        executeStatement("CREATE JAVA SCALAR SCRIPT " + fullyQualifiedScriptName + "() RETURNS INTEGER AS\n" //
+                + "    " + JAR_INCLUDE_DIRECTIVE + ";\n" //
+                + "    %scriptclass com.exasol.test.testobject.GetSizeUdf;\n" //
+                + "/\n\n");
```
Comment (kaklakariada) [inline src/test/java/com/exasol/test/JavaUdfIT.java:175]:
```suggestion
                + "    %scriptclass com.exasol.test.testobject.GetSizeUdf;");
```

File: src/test/java/com/exasol/test/JavaUdfIT.java [hunk 5]
```diff
@@ -150,13 +151,39 @@ void testGetTimestampFromSetScript() {
...
+        assertQueryResult("SELECT "+ fullyQualifiedScriptName + "()", table().row(1L));
+    }
+    @Test
+    void testGetSizeFromSetScript() {
+        final String scriptName = "SIZE_IN_SET_CONTEXT";
+        final String fullyQualifiedScriptName = getFullyQualifiedScriptName(schema, scriptName);
+        executeStatement("CREATE JAVA SET SCRIPT " + fullyQualifiedScriptName + "(COL CHAR(1)) RETURNS INTEGER AS\n" //
+                + "    " + JAR_INCLUDE_DIRECTIVE + ";\n" //
+                + "    %scriptclass com.exasol.test.testobject.GetSizeUdf;\n" //
+                + "/\n\n");
```
Comment (kaklakariada) [inline src/test/java/com/exasol/test/JavaUdfIT.java:185]:
```suggestion
                + "    %scriptclass com.exasol.test.testobject.GetSizeUdf;");
```

File: src/test/java/com/exasol/test/JavaUdfIT.java [hunk 6]
```diff
@@ -150,13 +151,39 @@ void testGetTimestampFromSetScript() {
...
+    void testGetSizeFromScalarScript() {
+        final String scriptName = "SIZE_IN_SCALAR_CONTEXT";
+        final String fullyQualifiedScriptName = getFullyQualifiedScriptName(schema, scriptName);
+        executeStatement("CREATE JAVA SCALAR SCRIPT " + fullyQualifiedScriptName + "() RETURNS INTEGER AS\n" //
+                + "    " + JAR_INCLUDE_DIRECTIVE + ";\n" //
+                + "    %scriptclass com.exasol.test.testobject.GetSizeUdf;\n" //
+                + "/\n\n");
+        assertQueryResult("SELECT "+ fullyQualifiedScriptName + "()", table().row(1L));
+    }
+    @Test
```
Comment (kaklakariada) [inline src/test/java/com/exasol/test/JavaUdfIT.java:178]:
```suggestion

    @Test
```

File: src/test/java/com/exasol/test/testobject/GetSizeUdf.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import com.exasol.ExaIterator;
+import com.exasol.ExaMetadata;
+
+/**
+ * This class implements a UDF function that is used in the integration tests. In this case we exercise the
+ * <code>size()</code> method.
+ * <p>
+ * The class mainly exists to force building against the API provided in this project. If you wrote inline UDFs, they
+ * would be compiled <i>inside</i> the database, using the API that is provided in the language container instead of
+ * this one. That means that you would not get compiler errors even if the APi in this project is broken (e.g. when
```
Comment (kaklakariada) [inline src/test/java/com/exasol/test/testobject/GetSizeUdf.java:12]:
```suggestion
 * this one. That means that you would not get compiler errors even if the API in this project is broken (e.g. when
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/test/JavaUdfIT.java (hunk 4 above)
Comment (redcatbear) [reply src/test/java/com/exasol/test/JavaUdfIT.java:175]: We need at least one newline or the UDF engine complains.

File: src/test/java/com/exasol/test/JavaUdfIT.java (hunk 5 above)
Comment (redcatbear) [reply src/test/java/com/exasol/test/JavaUdfIT.java:185]: Same here.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**